File: server/s2game/src/main/java/com/spectopics/s2game/dto/serverPayloads/CreatureBurnedPayload.java
package com.spectopics.s2game.dto.serverPayloads;

import com.spectopics.s2game.models.Creature;

public class CreatureBurnedPayload {
    // Tells the clients that a creature has been burned, and how much damage it took.
    public Creature creature;
    public float damage;

    public CreatureBurnedPayload(Creature creature, float damage) {
        this.creature = creature;
        this.damage = damage;
    }
}

File: front-end/src/dto/serverPayloads/CreatureBurnedPayload.js
import { Creature } from "../../models/Creature";

export class CreatureBurnedPayload {

    constructor(creature, damage) {
        this.creature = creature
        this.damage = damage
    }

    static fromDict(data) {
        return new CreatureBurnedPayload(
            Creature.fromDict(data.creature),
            data.damage
        );
    }   
}